test/test_data/test_feature_validator.py

```diff
@@ -237,9 +236,10 @@ def test_featurevalidator_categorical_nan(input_data_featuretest):
     validator.fit(input_data_featuretest)
     transformed_X = validator.transform(input_data_featuretest)
     assert any(pd.isna(input_data_featuretest))
-    assert any((-1 in categories) or ('-1' in categories) or ('Missing!' in categories) for categories in
-               validator.encoder.named_transformers_['encoder'].categories_)
-    assert np.shape(input_data_featuretest) == np.shape(transformed_X)
+    categories_ = validator.column_transformer.\
+        named_transformers_['categorical_pipeline'].named_steps['onehotencoder'].categories_
+    assert any(('0' in categories) or (0 in categories) or ('missing_value' in categories) for categories in
```

Comment:
This are the unique category values for the one hot encoder right, what would be the case where they are 0, is it not 'missing_value' for categorical columns?

Reply:
according to this https://github.com/automl/Auto-PyTorch/blob/10aea660389c899e9ab16410e95371c7c97f2436/autoPyTorch/data/tabular_feature_validator.py#L380-L381

It will be a categorical column only when it has a `string` or `bool` dtype.
And if between test and train there are differences, it will either throw an error or convert a null column to object for test if not empty in train.

Reply:
Does it fail if you run it without the `0` checks?

Reply:
> It will be a categorical column only when it has a `string` or `bool` dtype

not string, but category. And category can be string or int.

Reply:
It works fine without `'0'` check but removing `0` gives an error